pilot/cmd/pilot-agent/status/ready/probe_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ready
+
+import (
+	. "github.com/onsi/gomega"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+var probe = Probe{AdminPort: 1234}
```

Comment:
I'm not sure this is the right way to test the prober. Can you check if the e2e tests have the prober enabled ( they should), and maybe add pod/configs that may repro the problem ?

Reply:
(can be separate PR - not meant as a hold).

Reply:
Honestly, I think this is a perfect example for unit testing. No need for an e2e test imho. But actually, we are currently working on a e2e test for our scenario anyway. It should succeed in 1.0, but fail in the current master. So the probe would be active. This should safeguard our scenario. But since we are not familiar with the e2e framework, it might still take a bit. Therefor the plan was to put it in a separate PR. We will reference it, probably early next week.